synapse/handlers/sync.py

```diff
@@ -1630,6 +1618,22 @@ async def handle_room_entries(room_entry: "RoomSyncResultBuilder") -> None:
             newly_left_users,
         )
 
+    async def _get_ignored_users(self, user_id: str) -> FrozenSet[str]:
+        ignored_account_data = (
+            await self.store.get_global_account_data_by_type_for_user(
+                AccountDataTypes.IGNORED_USER_LIST, user_id=user_id
+            )
+        )
```

Comment:
I wonder if we can use the `ignored_users` table instead of this...? But that's a real change, not just a refactor.

Reply:
Didn't know about that, thanks. Though it's frustrating that it's not clear where the source of truth is. I'll add a #TODO.

Reply:
They should be in-sync, this table was added as an optimization. It might not take into account everything you need though!

Reply:
Note upcoming in ecccfb405.